compiler/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -85,8 +85,16 @@ impl Ssa {
 
         let mut brillig = Brillig::default();
 
+        if brillig_reachable_function_ids.is_empty() {
+            return brillig;
+        }
+
+        let func_id = brillig_reachable_function_ids
```

Comment:
What is this `func_id`? Is it [this](https://github.com/noir-lang/noir/blob/c8d5ce5e72f3171f24e4fe2b94ac098ba6d99ef6/compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs#L76C1-L80C42) `Function`?

Reply:
No it is just the first function from `brillig_reachable_function_ids`. The globals graph is computed a single time and shared among all functions https://github.com/noir-lang/noir/pull/7040 so we can use any func id here. I switched to using `self.main_id` to be more explicit.